src/main/terrier.cpp

```diff
@@ -1,53 +1,9 @@
-#include <network/terrier_server.h>
-#include <fstream>
-#include <iostream>
 #include <memory>
-#include <vector>
-#include "bwtree/bwtree.h"
-#include "common/allocator.h"
-#include "common/stat_registry.h"
-#include "common/strong_typedef.h"
-#include "loggers/index_logger.h"
-#include "loggers/main_logger.h"
-#include "loggers/network_logger.h"
-#include "loggers/parser_logger.h"
-#include "loggers/storage_logger.h"
-#include "loggers/transaction_logger.h"
-#include "loggers/type_logger.h"
-#include "storage/data_table.h"
-#include "storage/record_buffer.h"
-#include "storage/storage_defs.h"
-#include "transaction/transaction_context.h"
+#include "main/db_main.h"
 
-int main() {
-  // initialize loggers
-  try {
-    init_main_logger();
-    // initialize namespace specific loggers
-    terrier::storage::init_index_logger();
-    terrier::storage::init_storage_logger();
-    terrier::transaction::init_transaction_logger();
-    terrier::parser::init_parser_logger();
-    terrier::network::init_network_logger();
-    // Flush all *registered* loggers using a worker thread.
-    // Registered loggers must be thread safe for this to work correctly
-    spdlog::flush_every(std::chrono::seconds(DEBUG_LOG_FLUSH_INTERVAL));
-  } catch (const spdlog::spdlog_ex &ex) {
-    std::cout << "debug log init failed " << ex.what() << std::endl;  // NOLINT
-    return 1;
-  }
-  // log init now complete
-  LOG_TRACE("Logger initialization complete");
-
-  // initialize stat registry
-  auto main_stat_reg = std::make_shared<terrier::common::StatisticsRegistry>();
-
-  LOG_INFO("Initialization complete");
-
-  terrier::network::TerrierServer terrier_server;
-  terrier_server.SetupServer().ServerLoop();
-
-  // shutdown loggers
-  spdlog::shutdown();
-  main_stat_reg->Shutdown(false);
+int main(int argc, char **argv) {
+  auto db_main = std::make_shared<terrier::DBMain>();
+  db_main->Init(argc, argv);
+  db_main->Start();
```

Comment:
It should be made clear that `Start()` never returns in the normal case. When and how `Shutdown()` runs need to be documented.